### DISCUSSION TITLE
Hot reload behaviour

### DISCUSSION THREAD

**Discussion by sgcullen** [wxt-dev/wxt#1276] [General]
Is it possible to get hot reload working such that a manual browser refresh is not required?

**Comment by 1natsu172** [comment]
@sgcullen Could you give us a little more context? WXT has basic reloading feature.

**Reply by sgcullen** [reply]
Thanks @1natsu172 . I'm really just meaning that it would be great if chrome would self refresh on save rather than having to manually refresh the open dev chrome tab.

**Reply by aiktb** [reply]
> Thanks @1natsu172 . I'm really just meaning that it would be great if chrome would self refresh on save rather than having to manually refresh the open dev chrome tab.

WXT content script HMR has always worked for me....
What do you mean?

**Reply by sgcullen** [reply]
I think that's it @1natsu172 I'm using a Shadow Root and it looks like HMR isn't supported in that context:
https://wxt.dev/guide/essentials/content-scripts.html#ui

**Reply by 1natsu172** [reply]
@sgcullen 
refs: https://wxt.dev/guide/essentials/es-modules.html#content-scripts
> WXT does not yet include built-in support for ESM content scripts. The plan is to add support for chunking to reduce bundle size, but not support HMR for now. There are several technical issues that make implementing a generic solution for HMR impossible.

That's true for ShadowUI（content-script). But Live reload(tab reloading) should work. Does it not work too?

**Reply by sgcullen** [reply]
Thanks @1natsu172 
Tab reloading does indeed work so it's pretty good :-)
Thanks for explaining the limitation - that makes sense!
